--- a/op_builder/builder.py
+++ b/op_builder/builder.py
@@ -864,7 +864,7 @@
         if sys.platform == "win32":
             return ['-O2']
         else:
-            cpp_standard = '-std=c++20' if (TORCH_MAJOR, TORCH_MINOR) >= (2, 12) else '-std=c++17'
+            cpp_standard = '-std=c++20' if (TORCH_MAJOR, TORCH_MINOR) >= (2, 12) else ("-std=c++20" if tuple(map(int, torch.__version__.split(".")[:2])) >= (2, 13) else "-std=c++17")
             return ['-O3', cpp_standard, '-g', '-Wno-reorder']
 
     def nvcc_args(self):
@@ -874,7 +874,7 @@
         if self.is_rocm_pytorch():
             ROCM_MAJOR, ROCM_MINOR = self.installed_rocm_version()
             args += [
-                '-std=c++17', '-U__HIP_NO_HALF_OPERATORS__', '-U__HIP_NO_HALF_CONVERSIONS__',
+                ("-std=c++20" if tuple(map(int, torch.__version__.split(".")[:2])) >= (2, 13) else "-std=c++17"), '-U__HIP_NO_HALF_OPERATORS__', '-U__HIP_NO_HALF_CONVERSIONS__',
                 '-U__HIP_NO_HALF2_OPERATORS__',
                 '-DROCM_VERSION_MAJOR=%s' % ROCM_MAJOR,
                 '-DROCM_VERSION_MINOR=%s' % ROCM_MINOR
@@ -893,7 +893,7 @@
                 if cuda_major == 12 and cuda_minor >= 5:
                     std_lib = '-std=c++20'
                 else:
-                    std_lib = '-std=c++17'
+                    std_lib = ("-std=c++20" if tuple(map(int, torch.__version__.split(".")[:2])) >= (2, 13) else "-std=c++17")
             else:
                 std_lib = '-std=c++14'
             args += [
